### ISSUE TITLE
Feature Request: combination of bibm ads-search and bibm ads-add

### ISSUE THREAD

**Issue by AaronDavidSchneider** [pcubillos/bibmanager#65]
Thanks again for this incredible tool! Its in my daily use for organising my references.

My usual way to add a paper is the following:
`bibm ads-search`
`bibm ads-add` (I always use AuthorYear as key)
`bibm open` (to fetch the pdf)

I have a quick thought:
Why not link these commands? Maybe by using a format specifier (e.g. AuthorYear) for the citekey?
If overwriting `ads-search` isn't feasable, then maybe add a new command, like `ads-search-add`

What do you think? 🤔 

Again: Very nice tool!

**Comment by pcubillos** [comment]
Hi Aaron,
Thank you very much for your words! I'm glad that this is of use to others :)

I have two thoughts about this:

1.- Linking `ads-add` and `open` should be no problem (a lot of times I also wished I could run these two at once). IMO the simplest option would be to add a `-open` or `-fetch` flag to `ads-add`.  So, in my workflow after an `ads-search` I'd do this:
``` shell
bibm ads-add '2018A&A...617L...7S' Schneider2018
bibm fetch Schneider2018
```
The idea would be to have something like this:
``` shell
bibm ads-add -open '2018A&A...617L...7S' Schneider2018
```
or:
``` shell
bibm ads-add -fetch '2018A&A...617L...7S' Schneider2018
```
This is feasible, and I don't think very complicated to implement.  Does this sound like what you are looking for?


2.- Linking all three commands is more tricky, because the ads-search could return more than one result. Are you thinking of a Google-style "I'm feeling lucky" kind of workflow? Also, in general I try to avoid any scenario that needs to default an entry's key, because this is a very opinionated matter. I don't know, I feel like you need to know the output of an `ads-search` before deciding what to do next. I don't see how this could work, but if you have a more clear idea of the user-inputs, let me know :)

Thanks!,
p

**Comment by AaronDavidSchneider** [comment]
Hey,

thanks for the reply and for citing my paper 😊

I totally agree! Step 1 would already be very useful.

Regarding part 2:
I thought more about prompting. Like this:
`bibm ads-search-add` results in
```
(Press 'tab' for autocomplete)                                                  
author:"^Cubillos, Patricio" year:2018                                          

Title: Mass-loss rate constraints on the observed distribution of exoplanets
Authors: Cubillos, Patricio; et al.
adsurl:  https://ui.adsabs.harvard.edu/abs/2018EPSC...12..769C
bibcode: 2018EPSC...12..769C

Title: Open-source Software for Exoplanet Atmospheric Modeling
Authors: Cubillos, Patricio; et al.
adsurl:  https://ui.adsabs.harvard.edu/abs/2018AAS...23143918C
bibcode: 2018AAS...23143918C
```
which then prompts for example:
```
Do you want to add a paper? [Y/N/key]
```
where you can directly paste a key from above.

Something like this.

**Comment by pcubillos** [comment]
;)   Aha, I see now. Yeah, that makes sense, I like it.  And I think it's doable.
Thanks for the feedback!  I'll try to find some time to make this work.

**Comment by AaronDavidSchneider** [comment]
Awesome! Thanks.

**Comment by pcubillos** [comment]
OK!, Aaron, it took me some more thinking than I expected, but now one can add entries and/or fetch-open PDFs directly after an ADS search.
I updated the docs with a couple of examples (https://bibmanager.readthedocs.io/en/latest/ads.html#ads-search).

You can fetch/open PDFs even if they are not in your database.
Thanks for a great suggestion.

Resolved by b88f3fb

**Comment by AaronDavidSchneider** [comment]
Thanks! I will try ASAP.
Looks very good from the documentation.
